src/extension/mongodb/src/main/java/org/geoserver/rest/service/MongoStoreRestController.java

```diff
@@ -0,0 +1,126 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.service;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.lang3.StringUtils;
+import org.geoserver.catalog.DataStoreInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.rest.AbstractGeoServerController;
+import org.geoserver.rest.RestBaseController;
+import org.geotools.data.DataAccess;
+import org.geotools.data.complex.AppSchemaDataAccess;
+import org.geotools.data.complex.FeatureTypeMapping;
+import org.geotools.data.mongodb.MongoDataStore;
+import org.geotools.feature.NameImpl;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.Feature;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.Name;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Rest controller for handling operations over MongoDB datastores. */
+@RestController
+@ControllerAdvice
+@RequestMapping(
+    path = RestBaseController.ROOT_PATH + "/workspaces/{workspaceName}",
+    produces = {
+        MediaType.APPLICATION_JSON_VALUE,
+        MediaType.TEXT_HTML_VALUE,
+        MediaType.APPLICATION_XML_VALUE
+    }
+)
+public class MongoStoreRestController extends AbstractGeoServerController {
```

Comment:
Actually the REST endpoint clear persisted schemas and in-memory stored entries, since Mongo store checks and rebuild its schemas on demand (now using new parameters).